### PR TITLE
mdz 1.31.0 (new formula)

### PR DESCRIPTION
**Pull request by maxwelbm**
…Z is an open-source ledger CLI tool for managing and interacting with APIs.\n\nThis formula builds the binary from source using Go and injects required environment variables into the build via ldflags. It includes a test to validate the installation and version command.\n\nRepository: https://github.com/LerianStudio/midaz\nVersion: 1.31.0\nLicense: Apache-2.0

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/mdz.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.ClientID=\
+      9670e0ca55a29a466d31
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.ClientSecret=\
+      dd03f916cacf4a98c6a413d9c38ba102dce436a9
```
Comment (chenrui333) [inline Formula/m/mdz.rb:16]: what are these crypto values?

File: Formula/m/mdz.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.URLAPIAuth=\
+      http://127.0.0.1:8080
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.URLAPILedger=\
+      http://127.0.0.1:3000
+      -X github.com/LerianStudio/midaz/components/mdz/pkg/environment.Version=\
+      #{version}
+    ]
+
+    system "go", "build", *std_go_args(ldflags:), "./components/mdz/main.go"
```
Comment (chenrui333) [inline Formula/m/mdz.rb:25]:
```suggestion
    system "go", "build", *std_go_args(ldflags:), "./components/mdz"
```

File: Formula/m/mdz.rb (hunk 1 above)
Comment (maxwelbm) [reply Formula/m/mdz.rb]: These are just example values, the command being tested configure sets them to use cli as well

**Review by chenrui333 — APPROVED** [review]

**Review by daeho-ro — APPROVED** [review]

**Comment by chenrui333** [comment]
@maxwelbm, thanks for your first contribution to Homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**